speakeasy/binemu.py

```diff
@@ -1098,3 +1098,20 @@ def add_instruction_hook(self, cb, begin=1, end=0, ctx=[], emu=None, insn=None):
             hook.add()
 
         return hook
+
+    def add_invalid_instruction_hook(self, cb, ctx=[], emu=None):
+        if not emu:
+            emu = self
+
+        hook = common.InvalidInstructionHook(emu, self.emu_eng, cb, ctx=[])
+        hl = self.hooks.get(common.HOOK_INSN)
```

Comment:
copy-pasta error here.

more generally, i'd recommend considering unit tests that demonstrate how each feature works and assert that they behave as expected.

Reply:
```suggestion
        hl = self.hooks.get(common.HOOK_INSN_INVALID)
```